Add page metadata

laurapacilio opened the pull request:
**Description**
Updating the page description metadata on Introduction to Terraform pages, as part of this effort to [update all of the doc pages' metadata](https://app.asana.com/0/1128567758154570/1199892904127584/f) on terraform.io.

I also adjusted the copy on the Introduction to Terraform page. In the future, I will do this kind of thing in a separate PR. I tried to make the copy more concise and readable. 

Requested review scope:
- [x] Content touched by the PR only (typos, clarifications, tips)
- [ ] Code test (command and code block changes)
- [x] Flow and language near changes (new/rearranged steps)
- [ ] Review everything (rewrites, major changes)

Review urgency:
- [ ] ASAP (bug fixes, broken content, imminent releases)
- [x] 3 days (small changes, easy reviews)
- [ ] 1 week (default)
- [ ] Best effort (very non-urgent)

I have:
- [x] Verified that all status checks have passed
- [x] Verified that preview environment has successfully deployed.
- [x] Assigned an appropriate label (either Platform or a product name)
- [x] Requested at least one review

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/terraform?pullRequest=29051)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/terraform?pullRequest=29051)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/terraform?pullRequest=29051).</sub>

laurapacilio commented on website/intro/use-cases.html.markdown:
```diff
@@ -15,16 +15,11 @@ can be added to further extend Terraform's ability to manipulate resources.
 
-Heroku is a popular PaaS for hosting web apps. Developers create an app, and then
-attach add-ons, such as a database, or email provider. One of the best features is
+Heroku is a popular PaaS for hosting web apps. Developers create an app, and then attach add-ons, such as a database, or email provider. One of the best features is
```
Oh I think this was just like line length spacing (I think something weird happened with Atom and it changed that - there used to be hard breaks between lines. I didn't actually change anything :-)

laurapacilio commented on website/intro/vs/chef-puppet.html.markdown:
```diff
@@ -3,7 +3,7 @@ layout: "intro"
 sidebar_current: "vs-other-chef"
 description: |-
-  Configuration management tools install and manage software on a machine that already exists. Terraform is not a configuration management tool, and it allows existing tooling to focus on their strengths: bootstrapping and initializing resources.
+  An overview of how Terraform is different from configuration management tools.
```
Yes - thank you - that makes sense!

laurapacilio commented on website/intro/use-cases.html.markdown:
```diff
@@ -87,11 +72,8 @@ allows the network to better support the applications running on top. Most SDN
-versioned and changes to be automated. As an example, [AWS VPC](https://aws.amazon.com/vpc/)
-is one of the most commonly used SDN implementations, and [can be configured by
-Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/vpc).
+This configuration can then be used by Terraform to automatically setup and modify settings by interfacing with the control layer. This allows configuration to be
+versioned and changes to be automated. As an example, [AWS VPC](https://aws.amazon.com/vpc/) is one of the most commonly used SDN implementations, and [can be configured by Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/vpc).
```
Judith said I should review content in a separate PR to separate concerns, so that's why I didn't adjust this copy. (I did the initial "what is Terraform" page before we had that convo). But I do agree with you! Let me go back in and edit this page because I agree it needs a refresh.

laurapacilio commented on website/intro/index.html.markdown:
```diff
@@ -3,79 +3,44 @@ layout: "intro"
-
-> **Hands-on:** Try the [Get Started](https://learn.hashicorp.com/tutorials/terraform/infrastructure-as-code) tutorials on HashiCorp Learn.
-
 ## What is Terraform?
```
Yeah that's fair - I went back and forth on this - I'll remove it so it's shorter!

laurapacilio commented on website/intro/index.html.markdown:
```diff
@@ -3,79 +3,44 @@ layout: "intro"
 ## What is Terraform?
 
-Terraform is a tool for building, changing, and versioning infrastructure
-safely and efficiently. Terraform can manage existing and popular service
-providers as well as custom in-house solutions.
```
Got it - added that back in. See what you think!

im2nguyen reviewed (approved):
🚀

judithpatudith commented on website/intro/index.html.markdown:
```diff
@@ -3,79 +3,42 @@ layout: "intro"
-Infrastructure is described using a high-level configuration syntax. This allows
-a blueprint of your datacenter to be versioned and treated as you would any
-other code. Additionally, infrastructure can be shared and re-used.
+Infrastructure is described using a high-level [configuration language](/docs/language/index.html) in human-readable, declarative configuration files. This allows you to create a blueprint that can be versioned, shared, and reused.
```
Active vice 

- is described
- can be versioned ...

laurapacilio replied:
Thank you! fixed in my recent commit!

judithpatudith commented on website/intro/index.html.markdown:
```diff
@@ -3,79 +3,42 @@ layout: "intro"
-Terraform has a "planning" step where it generates an _execution plan_. The
-execution plan shows what Terraform will do when you call apply. This lets you
-avoid any surprises when Terraform manipulates infrastructure.
+Terraform generates an _execution plan_ describing what it will do and asks for your approval before creating, updating, or destroying infrastructure. This allows you to review changes before they are applied.
```
Active voice (are applied)

laurapacilio replied:
Also fixed!

judithpatudith commented on website/intro/index.html.markdown:
```diff
@@ -3,79 +3,42 @@ layout: "intro"
-safely and efficiently. Terraform can manage existing and popular service
-providers as well as custom in-house solutions.
-
-Configuration files describe to Terraform the components needed to
-run a single application or your entire datacenter.
-Terraform generates an execution plan describing
-what it will do to reach the desired state, and then executes it to build the
-described infrastructure. As the configuration changes, Terraform is able
-to determine what changed and create incremental execution plans which
-can be applied.
```
I think we lost the bit about Terraform comparing the infrastructure to the configuration to make changes. Maybe we can add that bit to the change automation section?

laurapacilio replied:
Good call - thank you for flagging. I added that into that section - let me know what you think.

judithpatudith commented on website/intro/index.html.markdown:
```diff
@@ -3,79 +3,42 @@ layout: "intro"
-and modification of any non-dependent resources. Because of this, Terraform
-builds infrastructure as efficiently as possible, and operators get insight into
-dependencies in their infrastructure.
+Terraform builds a resource graph and parallelizes the creation and modification of any non-dependent resources. This allows Terraform to
```
Maybe "creates or modifies non-dependent resources in parallel"?

laurapacilio replied:
yes! Changed that in my commit.

judithpatudith commented on website/intro/index.html.markdown:
```diff
@@ -3,79 +3,42 @@ layout: "intro"
-builds infrastructure as efficiently as possible, and operators get insight into
-dependencies in their infrastructure.
+Terraform builds a resource graph and parallelizes the creation and modification of any non-dependent resources. This allows Terraform to
+build resources as efficiently as possible and gives operators greater insight into their infrastructure.
```
```suggestion
build resources as efficiently as possible and gives you greater insight into your infrastructure.
```

laurapacilio replied:
Done! I wasn't sure about this one because I went back and forth about whether the audience might be broader somehow because this is an overview page...but then I realized that it's likely operators who are this deep in reference docs. Thank you!

judithpatudith commented on website/intro/use-cases.html.markdown:
```diff
@@ -3,28 +3,21 @@ layout: "intro"
 sidebar_current: "use-cases"
 description: |-
-  Before understanding use cases, it's useful to know what Terraform is. This page lists some concrete use cases for Terraform, but the possible use cases are much broader than what we cover. Due to its extensible nature, providers and provisioners can be added to further extend Terraform's ability to manipulate resources.
+  Learn common use cases for Terraform including Heroku apps, self-service clusters, and multi-cloud deployment. 
```
```suggestion
  Learn common use cases for Terraform including managing Heroku apps, self-service clusters, and multi-cloud deployments. 
```

judithpatudith commented on website/intro/use-cases.html.markdown:
```diff
@@ -3,28 +3,21 @@ layout: "intro"
-Before understanding use cases, it's useful to know [what Terraform is](/intro/index.html).
-This page lists some concrete use cases for Terraform, but the possible use cases are
-much broader than what we cover. Due to its extensible nature, providers and provisioners
+This page lists a subset of use cases for [Terraform](/intro/index.html). Due to its extensible nature, providers and provisioners
```
I think the second sentence is talking about building your own custom providers, but it kind of makes it sound like you only need a provider or provisioner if you want to do more with Terraform than the binary does alone.

I think we could either delete this line since this is beginner content and writing plugins is advanced, or just clarify that you can write your own Terraform plugins if you have a use case that the currently available ones don't cover.

laurapacilio replied:
Done - I've removed the sentence. I agree - we don't need to go into that here, since this is intro stuff.

judithpatudith commented on website/intro/vs/boto.html.markdown:
```diff
@@ -3,7 +3,7 @@ layout: "intro"
 sidebar_current: "vs-other-boto"
 description: |-
-  Libraries like Boto, Fog, etc. are used to provide native access to cloud providers and services by using their APIs. Some libraries are focused on specific clouds, while others attempt to bridge them all and mask the semantic differences. Using a client library only provides low-level access to APIs, requiring application developers to create their own tooling to build and manage their infrast ...
+  How Terraform compares to cloud provider client libraries. 
```
We should keep specific names of technologies that the page covers in the descriptions so that search engines will bring this page up when people search for them.

laurapacilio replied:
Ah! That makes perfect sense. I'll go back through and make sure the technologies are added back in everywhere.

judithpatudith reviewed (changes requested):
This is great, thanks! 

Some specific comments in line, but one thing I noticed for all of the tech comparison pages is that we definitely want to keep the specific technology names in the page descriptions. That way, search engines will surface them when people search for "Terraform vs X". In fact, we probably want to add the names of any tools that the pages cover but weren't in the descriptions before :)

laurapacilio commented:
Okay @judithpatudith I think I addressed everything you flagged:
- Active voice on intro page
- Accepted your suggestions on wording in a few places
- Reworked the descriptions for vs so that the product names are back + a few extra! 

Thank you for the feedback! Please let me know if you see anything else :)

judithpatudith reviewed (approved):
Looks great, thank you!
